src/reference/asciidoc/ws.adoc

```diff
@@ -196,3 +200,38 @@ Negated patterns get priority, so a list such as
 
 IMPORTANT: If you have a user defined header that begins with `!` that you *do* wish to map, you need to escape it with
 `\` thus: `STANDARD_REQUEST_HEADERS,\!myBangHeader` and it *WILL* be mapped.
+
+Inbound SOAP headers (request headers for the inbound gateway, reply-headers for the outbound gateway) are mapped as
+`SoapHeaderElement` objects.
+The contents can be explored by accessing the `Source`:
+
+[source, xml]
+----
+<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+    <soapenv:Header>
+        <auth>
+            <username>user</username>
+            <password>pass</password>
+        </auth>
+        <bar>BAR</bar>
+        <baz>BAZ</baz>
+        <qux>qux</qux>
+    </soapenv:Header>
+    <soapenv:Body>
+        ...
+    </soapenv:Body>
+</soapenv:Envelope>
+----
+
+If `mapped-request-headers` is `"auth, ba*"`, the `auth`, `bar` and `baz` headers are mapped but `qux` is not.
+
+[source, java]
+----
+...
+SoapHeaderElement header = (SoapHeaderElement) headers.get(authHeader);
```

Comment:
I understand that it is a copy/paste from the test, but here in the doc it isn't clear what is `authHeader` variable.

Will fix on merge to the just `"auth"` String.